index.html

```diff
@@ -854,53 +750,159 @@ <h2>
+          that represents now.
+          </li>
+          <li>Let |timeoutTime| be the sum of |acquisitionTime| and
+          |options|.{{PositionOptions/timeout}}.
           </li>
-          <li data-cite="permissions">If |permission| is
-          {{PermissionState/"denied"}}, then:
+          <li>Let |cachedPosition:GeolocationPosition| be [=this=]'s
+          {{Geolocation/[[cachedPosition]]}}.
+          </li>
+          <li>Create an implementation-specific |timeout| task that elapses at
+          |timeoutTime|, during which it tries to acquire the device's position
+          by running the following steps:
             <ol>
-              <li>[=Call back with error=] |errorCallback| and
-              {{GeolocationPositionError/PERMISSION_DENIED}}.
+              <li>Let |permission| be [=get the current permission state=] of
+              <a>"geolocation"</a>.
+              </li>
+              <li>If |permission| is "denied":
+                <ol>
+                  <li>Stop |timeout|.
+                  </li>
+                  <li>Do the <a href="#permission-deined">user or system denied
+                  permission</a> failure case step.
+                  </li>
+                </ol>
               </li>
-              <li>Return failure.
+              <li>If |permission| is "granted":
+                <ol>
+                  <li>Let |position| be null.
+                  </li>
+                  <li>If |cachedPosition| is not null, and
+                  |options|.{{PositionOptions/maximumAge}} is greater than 0:
+                    <ol>
+                      <li>Let |cacheTime:long| be |acquisitionTime| minus the
+                      value of the |options|.{{PositionOptions/maximumAge}}
+                      member.
+                      </li>
+                      <li>If |cachedPosition|'s
+                      {{GeolocationPosition/timestamp}}'s value is greater than
+                      |cacheTime|, and
+                      |cachedPosition|.{{GeolocationPosition/[[isHighAccuracy]]}}
+                      equals |options|.{{PositionOptions/enableHighAccuracy}},
+                      set |position| to |cachedPosition|.
+                      </li>
+                    </ol>
+                  </li>
+                  <li>Otherwise, if |position| is not |cachedPosition|, try to
+                  acquire position data from the underlying system, optionally
+                  taking into consideration the value of
+                  |options|.{{PositionOptions/enableHighAccuracy}} during
+                  acquisition.
+                  </li>
+                  <li>If the |timeout| elapses during acquisition, or acquiring
+                  the device's position results in failure:
+                    <ol>
+                      <li>Stop the |timeout|.
+                      </li>
+                      <li>Go to <a href="#errors-reasons">dealing with
```

Comment:
Queue a task to deal with these failures.

Reply:
Yep, already done below: "[=Call back with error=]" actually does the task queueing on the "geolocation task source".

Reply:
Ah, I missed that "call back with error" was a Geolocation API-specific algorithm and not just a generic one to invoke a callback.